### PR TITLE
feat: add test against live Node

### LINKED ISSUE #1281: Run integration tests against a live node
Mirror of https://github.com/FuelLabs/fuels-rs/issues/1135.

### PR DESCRIPTION
**Pull request by camsjams**
Closes #1281 

### Sample Test
See https://github.com/FuelLabs/fuels-ts/actions/runs/6439505793/job/17487267257?pr=1324

image

#### Wallet Information
- bech32 address: fuel1x33ajpj0jy5p2wcqqu45e32r75zrwfeh6hwqfv5un670rv4p0mns58enjg
- add assets: https://faucet-beta-4.fuel.network/
- check balance: https://fuellabs.github.io/block-explorer-v2/beta-4/#/address/fuel1x33ajpj0jy5p2wcqqu45e32r75zrwfeh6hwqfv5un670rv4p0mns58enjg

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -72,3 +72,30 @@ jobs:
...
+        run: pnpm build
+
+      - name: Run Isolated Tests
+        run: pnpm test:e2e
+        env:
+          FUEL_NETWORK_URL: https://beta-4.fuel.network/graphql
+          # not enough coins to fit the target?
+          # add assets: https://faucet-beta-4.fuel.network/
+          # bech32 address: fuel1x33ajpj0jy5p2wcqqu45e32r75zrwfeh6hwqfv5un670rv4p0mns58enjg
+          # check balance: https://fuellabs.github.io/block-explorer-v2/beta-4/#/address/fuel1x33ajpj0jy5p2wcqqu45e32r75zrwfeh6hwqfv5un670rv4p0mns58enjg
```
Comment (danielbate) [inline .github/workflows/test.yaml:100]: Would this be better placed in the live env specific test? Could have a try catch in there that logs this out?

**Review by danielbate — DISMISSED** [review]
:shipit:

File: package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "ci:test": "./scripts/ci-test.sh",
     "pretest": "turbo run pretest",
     "test": "jest --no-cache --runInBand --coverage",
+    "test:e2e": "jest --no-cache --runInBand packages/fuel-gauge/src/live-script.test.ts",
```
[the remark this answers is not in the document]
Comment (arboleya) [reply package.json]: I assume we all have personal derivations of how we call each test type and how each translates to our specific (Fuel) case, and I know @danielbate has been sorting this out and writing some specs around the topic, but I'll leave my current line of thought here, just in case.

 - `Unit` = Worries about units, usually mocking external stuff
 - `Feature` = Worries about features, not mocking or mocking minimally
 - `Integration` = Tests that target `fuel-core` locally
 - `E2E` - Tests that target `fuel-core` remotely

In other words, I am fond of using `e2e` for this case.

[the remark this answers is not in the document]
Comment (nedsalk) [reply package.json]: `test:live` seems like an options as well.

**Review by danielbate — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -72,3 +72,29 @@ jobs:
           test-script: pnpm ci:test
           prnumber: ${{ steps.findPr.outputs.number }}
           base-coverage-file: coverage-reports/report.master.json
+  live-tests:
+    name: Run Live Tests
+    # if: github.head_ref = 'changeset-release/master'
+    # temp for proof of concept on GH PR
+    if: github.ref == 'refs/pull/1324/merge'
```
Comment (arboleya) [inline .github/workflows/test.yaml:79]: Shouldn't we run this for every PR instead of just for release ones?

It could help block broken stuff from being merged into `master` and, even better, reduce the risk of remedying broken releases.

**Review by arboleya — DISMISSED** [review]

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by digorithm — APPROVED** [review]
Looking good. It's simple, but I think it gets the job done. We might want to add more test cases later, but this covers the basics.

(branch is outdated tho)

**Comment by arboleya** [comment]
@camsjams The test step for this one seems stuck since https://github.com/FuelLabs/fuels-ts/pull/1324/commits/80ba2a641ec93f91dfa8be1771337a2e31484c47.

**Review by arboleya — APPROVED** [review]

**Review by nedsalk — APPROVED** [review]
